main.go

```diff
@@ -224,37 +226,31 @@ func Unpin(b *hb.Bot, actor, path string) {
 	clusterPinUnpin(b, actor, path, "", false)
 }
 
+// StatusCluster gets cluster status of cid with given path.
 func StatusCluster(b *hb.Bot, actor, path string) {
 	ctx := context.Background()
-	// pick random cluster
-	i := rand.Intn(len(clusters))
-	cluster := clusters[i]
-	addr := clusterAddrs[i]
 
-	c, err := resolveCid(path, shs[i], shsUrls[i])
+	c, err := resolveCid(path, lbShell)
```

Comment:
I don't see that lbShell is ever set. This will likely panic when attempting to resolve an /ipns path.

Reply:
Also, this can be upgraded since cluster knows how to resolve paths now, so it should use PinPath etc.

Reply:
> Also, this can be upgraded since cluster knows how to resolve paths now, so it should use PinPath etc.

For cluster-pinning and cluster-unpining pin/unpin with path are used. For other operations such as status, recover etc, I have used recover cid.